### ISSUE TITLE
sgt constructor problem v. 1.2.13

### ISSUE THREAD

**Issue by Fefocrates** [TheBrownLab/PhyloFisher#126]
Hey all!

Im currently getting the following error messages from the sgt constructor

sgt_constructor.py -t 50 -i working_dataset_constructor_out_Nov.06.2024

Building DAG of jobs...
Creating conda environment /home/arodr412/mambaforge/envs/fisher/bin/mafft.yaml...
Downloading and installing remote packages.
CreateCondaEnvironmentException:
Could not create conda environment from /home/arodr412/mambaforge/envs/fisher/bin/mafft.yaml:
Traceback (most recent call last):
  File "/home/arodr412/mambaforge/condabin/mamba", line 7, in <module>
    from mamba.mamba import main
  File "/home/arodr412/mambaforge/lib/python3.10/site-packages/mamba/mamba.py", line 18, in <module>
    from conda.cli.main import generate_parser, init_loggers
ImportError: cannot import name 'generate_parser' from 'conda.cli.main' (/home/arodr412/mambaforge/lib/python3.10/site-packages/conda/cli/main.py)

  File "/home/arodr412/mambaforge/envs/fisher/lib/python3.7/site-packages/snakemake/deployment/conda.py", line 390, in create
Traceback (most recent call last):
  File "/home/arodr412/mambaforge/envs/fisher/bin/sgt_constructor.py", line 150, in <module>
    run_snakemake(length_filter=False)
  File "/home/arodr412/mambaforge/envs/fisher/bin/sgt_constructor.py", line 85, in run_snakemake
    f'--config {make_config(length_filter)}',
  File "/home/arodr412/mambaforge/envs/fisher/bin/sgt_constructor.py", line 49, in make_config
    f'genes={",".join(get_genes(length_filter))}',
  File "/home/arodr412/mambaforge/envs/fisher/bin/sgt_constructor.py", line 27, in get_genes
    bmge_out_files = [file for file in os.listdir(len_filt_bmge_dir) if file.endswith('.bmge')]
FileNotFoundError: [Errno 2] No such file or directory: '/home/arodr412/hokuleaphylo/hokuwatst/sgt_constructor_out_Nov.06.2024/length_filtration/bmge'

# Ok so import it manually???
## ImportError: cannot import name 'generate_parser' from 'conda.cli.main' 
/home/arodr412/mambaforge/lib/python3.10/site-packages/conda/cli/

(fisher) [arodr412@sc009:~/mambaforge/lib/python3.10/site-packages/conda/cli]$ ls
actions.py         install.py        main_env_config.py  main_env_update.py  main_list.py             main.py         __pycache__/
common.py          main_clean.py     main_env_create.py  main_env_vars.py    main_mock_activate.py    main_remove.py  python_api.py
conda_argparse.py  main_commands.py  main_env_export.py  main_export.py      main_mock_deactivate.py  main_rename.py
find_commands.py   main_compare.py   main_env_list.py    main_info.py        main_notices.py          main_run.py
helpers.py         main_config.py    main_env.py         main_init.py        main_package.py          main_search.py
__init__.py        main_create.py    main_env_remove.py  main_install.py     main_pip.py              main_update.py

## it was trying to access the directory which it couldn't create basically because of the reasons above?
 
FileNotFoundError: [Errno 2] No such file or directory: '/home/arodr412/hokuleaphylo/hokuwatst/sgt_constructor_out_Nov.06.2024/length_filtration/bmge'

Any help and input is appreciated!

**Comment by Fefocrates** [comment]
Its fixed!! downgrade the mamba version  " mamba install -c conda-forge mamba=1 "

**Comment by Edouard94** [comment]
Hello everyone,

I encountered the same problem as @Fefocrates but I also resolved the issue with reinstalling _PhyloFisher_ in my 'fisher' environment with:

`mamba install python=3.9`
`mamba install -c conda-forge mamba=1`
`mamba install python=3.7.10`
`mamba install phylofisher`

I don't really understand the issue of the error with the bmge, mafft and other modules, could it be linked to incompatibility between the .yaml files and our own conda environments? 

It seems to work now anyways.

Best wishes and thanks again for the tool!
